internal/mode/static/state/graph/graph.go

```diff
@@ -46,6 +48,8 @@ type Graph struct {
 	ReferencedSecrets map[types.NamespacedName]*Secret
 	// ReferencedNamespaces includes Namespaces with labels that match the Gateway Listener's label selector.
 	ReferencedNamespaces map[types.NamespacedName]*v1.Namespace
+	// ReferencedServicesNames includes the names of all the Services that are referenced by at least one HTTPRoute.
+	ReferencedServicesNames map[types.NamespacedName]struct{}
```

Comment:
I like `ReferencedServices` but I'm worried it might be confusing to use since it doesn't actually store the Service. `ReferencedSecrets` and `ReferencedNamespaces` both follow the same structure of storing a `NamespacedName` then the actual resource, but `ReferencedServices` wouldn't. 

Would it be alright for me to add a little comment specifying why storing a Service isn't necessary? e.g.

```
// ReferencedServices includes the NamespacedNames of all the Services that are referenced by at least one HTTPRoute.
// ReferencedServices does not store the Service resource as the NamespacedName of the Service is sufficient for the current use case.
ReferencedServicesNames map[types.NamespacedName]struct{}
```

Or do you think that is unnecessary?

Reply:
in my view, it is not necessary - if we needed to store the whole service, we would have stored. however, because other maps above store whole resources, ReferencedServices looks inconsistent, that could be confusing. For that, I suggest mentioning other maps in the comment
```
// ReferencedServices includes the NamespacedNames of all the Services that are referenced by at least one HTTPRoute.
// Storing the whole resource is not necessary, compared to the similar maps above.
ReferencedServicesNames map[types.NamespacedName]struct{}
```

However, this is not exact science, we can get confused by different or the same thing, and thus have different views. I don't see much harm in that comment if it helps you to understand the code better

Reply:
Okay yea sure that comment would be helpful to me and would help me understand the code better if I were reading it for the first time. Also, did you mean to keep the name change to `ReferencedServices` in your example?

Reply:
ReferencedServices 👍